### DISCUSSION TITLE
Representation for DISTINCT in aggregate functions

### DISCUSSION THREAD

**Discussion by JamesRTaylor** [substrait-io/substrait#188] [General]
What's the best way in the spec to represent DISTINCT usage in aggregate functions (i.e. COUNT(DISTINCT foo) )? There seem to be two options: specialized functions or an aggregate function invocation property. Any other ideas? Thoughts on which would be best?

**Comment by jacques-n** [comment]
Tagging @westonpace , @cpcloud, @andygrove, @hannes , @pitrou for feedback.

**Comment by andygrove** [comment]
In DataFusion we made it a property of invoking an aggregate function.

**Reply by cpcloud** [reply]
In principle any column whose values can be compared using `==` can be DISTINCT input to an aggregation, so I'm +1 on having it as a property of aggregate functions instead of having separate `{agg}_distinct` functions.

**Reply by westonpace** [reply]
I'm -1 on `{agg}_distinct`.  I don't understand the alternative enough to say.  When we say a property of invoking the aggregate function are we really talking about a property on the `AggregateFunction` message?  Or is this part of the `AggregateRel::Measure` messages?

**Reply by JamesRTaylor** [reply]
I was thinking it'd be a property on `AggregateFunction` message.

**Reply by JamesRTaylor** [reply]
For example, like this PR: https://github.com/substrait-io/substrait/pull/191

**Comment by hannes** [comment]
We have this as a flag `is_distinct` in the `BoundAggregateExpression` https://github.com/duckdb/duckdb/blob/master/src/include/duckdb/planner/expression/bound_aggregate_expression.hpp#L28

**Comment by jacques-n** [comment]
Note, this is now addressed via #191 and will be in the next release.
